src/auth.c

```diff
@@ -939,6 +939,9 @@ _handle_features_sasl(xmpp_conn_t *conn, xmpp_stanza_t *stanza, void *userdata)
     if (bind) {
         ns = xmpp_stanza_get_ns(bind);
         conn->bind_required = ns != NULL && strcmp(ns, XMPP_NS_BIND) == 0;
+        xmpp_stanza_clone(bind);
```

Comment:
I have concern about this approach. `_do_bind()` inserts `bind` argument to a stanza as a child, so it's `next` field is changed. But at the same time `bind` is a child of the features stanza. So looks like this will lead to few issues with releasing parent stanzas, children list corruption, etc.
Probably we still need to make a copy instead of clone, but I would like to hide this logic in `_do_bind()` to improve isolation.

Reply:
> [...] issues with releasing parent stanzas, children list corruption [...] we still need to make a copy instead of clone [...]

Indeed, I didn't think of that!

> I would like to hide this logic in `_do_bind()` to improve isolation

As we need the copied `bind` stanza in the "bind case" and the "resume case" I guess the best spot would be to do the copy where it is now, and not move it. Is that fine?

Reply:
Thanks a lot for the review!